client/sniffer/src/test/java/org/elasticsearch/client/sniff/ElasticsearchNodesSnifferParseTests.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Test parsing the response from the {@code /_nodes/http} API from fixed
+ * versions of Elasticsearch.
+ */
+public class ElasticsearchNodesSnifferParseTests extends RestClientTestCase {
+    private void checkFile(String file, Node... expected) throws IOException {
+        InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(file);
+        if (in == null) {
+            throw new IllegalArgumentException("Couldn't find [" + file + "]");
+        }
+        try {
+            HttpEntity entity = new InputStreamEntity(in, ContentType.APPLICATION_JSON);
+            List<Node> nodes = ElasticsearchNodesSniffer.readHosts(entity, Scheme.HTTP, new JsonFactory());
+            // Sort he list so the error messages are easier to read.
+            Collections.sort(nodes, new Comparator<Node>() {
+                @Override
+                public int compare(Node lhs, Node rhs) {
+                    return lhs.getName().compareTo(rhs.getName());
+                }
+            });
+            assertNodesEquals(Arrays.asList(expected), nodes);
+        } finally {
+            in.close();
+        }
+    }
+
+    public void test2x() throws IOException {
+        checkFile("2.0.0_nodes_http.json",
+                node(9207, "c1", "2.0.0", false, false, false),
+                node(9206, "c2", "2.0.0", false, false, false),
+                node(9205, "d1", "2.0.0", false, true, false),
+                node(9204, "d2", "2.0.0", false, true, false),
+                node(9203, "d3", "2.0.0", false, true, false),
+                node(9200, "m1", "2.0.0", true, false, false),
+                node(9202, "m2", "2.0.0", true, true, false),
+                node(9201, "m3", "2.0.0", true, false, false));
+    }
+
+    public void test5x() throws IOException {
+        checkFile("5.0.0_nodes_http.json",
+                node(9206, "c1", "5.0.0", false, false, true),
+                node(9207, "c2", "5.0.0", false, false, true),
+                node(9203, "d1", "5.0.0", false, true, true),
+                node(9204, "d2", "5.0.0", false, true, true),
+                node(9205, "d3", "5.0.0", false, true, true),
+                node(9200, "m1", "5.0.0", true, false, true),
+                node(9201, "m2", "5.0.0", true, true, true),
+                node(9202, "m3", "5.0.0", true, false, true));
+    }
+
+    public void test6x() throws IOException {
+        checkFile("6.0.0_nodes_http.json",
+                node(9206, "c1", "6.0.0", false, false, true),
+                node(9207, "c2", "6.0.0", false, false, true),
+                node(9203, "d1", "6.0.0", false, true, true),
+                node(9204, "d2", "6.0.0", false, true, true),
+                node(9205, "d3", "6.0.0", false, true, true),
+                node(9200, "m1", "6.0.0", true, false, true),
+                node(9201, "m2", "6.0.0", true, true, true),
+                node(9202, "m3", "6.0.0", true, false, true));
+    }
```

Comment:
nice, thanks for adding these!

Reply:
I took a couple of hours building them from real nodes downloaded from old version! Lovingly crafted.